internal/ethapi/api.go

```diff
@@ -1253,10 +1252,16 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		result.GasFeeCap = (*hexutil.Big)(tx.GasFeeCap())
 		result.GasTipCap = (*hexutil.Big)(tx.GasTipCap())
 		// if the transaction has been mined, compute the effective gas price
-		if baseFee != nil && blockHash != (common.Hash{}) {
-			// price = min(tip, gasFeeCap - baseFee) + baseFee
-			price := math.BigMin(new(big.Int).Add(tx.GasTipCap(), baseFee), tx.GasFeeCap())
-			result.GasPrice = (*hexutil.Big)(price)
+		if blockHash != (common.Hash{}) {
+			baseFee, _ := baseFeeFn(tx.FeeCurrency())
```

Comment:
it would be better if we use the error and then check `if err != nil || baseFee == nil`. Seems more robust